src/processors/ipfsFile/syncExistingPinsProcessor.ts

```diff
@@ -11,29 +12,80 @@ const ipfsFilesOutOfSyncWithPins: (field: TrackMediaField) => Trigger<undefined>
     select distinct "${field}IPFSHash", "${field}URL" from "${Table.processedTracks}" as t
     left outer join "${Table.ipfsFiles}" as file
     on t."${field}IPFSHash" = file.cid
-    where
+    where (
       t."${field}IPFSHash" is not null and
       file.cid is null and
       file.error is null and
       t."${field}URL" ILIKE ('%' || t."${field}IPFSHash" || '%')
+    ) or (
+      t."${field}IPFSHash" is not null and
+      t."${field}URL" is not null and
+      file.cid is null and
+      file.error is null
+    )
   `)
 
    return tracksWithCIDsMatchingURL.rows;
  }
 
+const ipfsFilesPinnedWithoutUrl: (field: TrackMediaField) => Trigger<undefined> =
+ (field) => async (clients) => {
+   const tracksWithCIDWithoutURL = await clients.db.rawSQL(`
+    select "id", "${field}IPFSHash", "${field}URL" from "${Table.processedTracks}" as t
+    left outer join "${Table.ipfsFilesUrls}" as url
+    on t."${field}URL" = url.url
+    where
+      t."${field}IPFSHash" is not null and
+      t."${field}URL" is null and
+      url.cid is null and
+      url.error is null
+  `)
+
+   return tracksWithCIDWithoutURL.rows;
+ }
+
 export const ipfsFileSyncExistingPinsProcessor: (field: TrackMediaField) => Processor =
 (field) => {
   return {
     name: 'ipfsFileSyncExistingPinsProcessor',
     trigger: ipfsFilesOutOfSyncWithPins(field),
-    processorFunction: async (input: IPFSFile[], clients: Clients) => {
+    processorFunction: async (input: any[], clients: Clients) => {
       console.log(`Adding ${field} ipfs files for media pinned by other platforms`);
 
-      const files = input.map((row: any) => ({
+      const files: IPFSFile[] = input.map((row: any) => ({ cid: row[`${field}IPFSHash`] }))
+      await clients.db.insert(Table.ipfsFiles, files);
+
+      const filesUrls: IPFSFileUrl[] = input.map((row: any) => ({
         url: row[`${field}URL`],
         cid: row[`${field}IPFSHash`],
       }))
-      await clients.db.insert(Table.ipfsFiles, files);
+      await clients.db.insert(Table.ipfsFilesUrls, filesUrls);
+    }
+  }
+}
+
+export const ipfsFileSyncExternalPinsProcessor: (field: TrackMediaField) => Processor =
```

Comment:
this processor is related to ExternalUploads, not ExternalPins - let's rename it and the trigger to refer to uploads rather than pins to avoid confusion